### PR TITLE
Add support for emitting functions with `coldcc` to LLVM

### PR DESCRIPTION
**Pull request by scottmcm**
The eventual goal is to try using this for things like the internal panicking stuff, to see whether it helps.

### CODE REVIEW

File: compiler/rustc_codegen_cranelift/src/abi/mod.rs [hunk 1]
```diff
@@ -22,7 +22,7 @@ fn clif_sig_from_fn_abi<'tcx>(
     fn_abi: &FnAbi<'tcx, Ty<'tcx>>,
 ) -> Signature {
     let call_conv = match fn_abi.conv {
-        Conv::Rust | Conv::C => default_call_conv,
+        Conv::Rust | Conv::C | Conv::RustCold => default_call_conv,
```
Comment (bjorn3) [inline compiler/rustc_codegen_cranelift/src/abi/mod.rs:25]: You could use `CallConv::Cold` here in principle. Cranelift currently implements it the same as SystemV currently though and I'm not sure how well it handles mixed calling conventions.

Comment (scottmcm) [reply compiler/rustc_codegen_cranelift/src/abi/mod.rs:25]: Oh, thanks! [Smallest caller code size](https://docs.rs/cranelift-codegen/latest/cranelift_codegen/isa/enum.CallConv.html#variant.Cold) sounds great; I should have looked earlier.

**Comment by lcnr** [comment]
impl looks good to me

r? @nagisa as I have no idea about llvm

**Comment by thomcc** [comment]
This is great to see! I've wanted this in libs for a while now.

**Comment by nagisa** [comment]
@rustbot author

Feel free to r=me after fixing the test failures.

**Comment by scottmcm** [comment]
@bors retry

This doesn't look like something that could have been caused by me:

> I/O error: operation failed to complete synchronously

**Pull request merged**